pkg/discovery/dtofactory/container_dto_builder.go

```diff
@@ -220,3 +241,14 @@ func (builder *containerDTOBuilder) addPodProperties(pod *api.Pod, index int) []
 
 	return properties
 }
+
+// Get the stitching property for Application.
+func getContainerStitchingProperty(pod *api.Pod, index int) string {
+	// Parse the container id from the status, it is either in the form of
+	// containerID: docker://986c8fc7247d1ff047f06e8e3487029b89baef1e908d3e334dd94aaa3bf4bc39
+	// or
+	// containerID: cri-o://81ab6978aa3252bc53cfaf7d23b8b1bf5f8f27ff4b9e79a0fbaffebeddaeeb4b
+	containerStatus := &(pod.Status.ContainerStatuses[index])
+	cntiduri := strings.Split(containerStatus.ContainerID, "://")
+	return cntiduri[1]
```

Comment:
Along the same line to avoid any possibility of index out of range panic, we could add a check here too e.g.
```
if len(cntiduri) <= 1 {
  return ""
}
```

Reply:
done, thanks